### PR TITLE
NLP-ENGINE-202 ICU Unicode integration

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Signed-off-by: David de Hilster <david.dehilster@lexisnexisrisk.com>

### CODE REVIEW

File: .github/workflows/test-platforms.yml [hunk 1]
```diff
@@ -7,21 +7,63 @@ on:
     types: [opened, synchronize, reopened]
 
 jobs:
-  Build:
+  job:
+    name: ${{ matrix.os }} - ${{ github.workflow }}
     runs-on: ${{ matrix.os }}
```
Comment (GordonSmith) [inline .github/workflows/test-platforms.yml:12]: Should probably be declared after the srategy.matrix declaration

File: .github/workflows/test-platforms.yml [hunk 2]
```diff
@@ -7,21 +7,63 @@ on:
...
+      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: windows-latest
+            triplet: x64-windows
+          - os: ubuntu-latest
+            triplet: x64-linux
+          - os: macos-latest
+            triplet: x64-osx
```
Comment (GordonSmith) [inline .github/workflows/test-platforms.yml:23]: The include order should probably match the matrix.os order

File: .github/workflows/test-platforms.yml [hunk 3]
```diff
@@ -7,21 +7,63 @@ on:
...
+        if: ${{ matrix.os == 'windows-latest' }}
+
+      - name: Vcpkg bootstrap Shell
+        run: ./bootstrap-vcpkg.sh
+        shell: bash
+        working-directory: vcpkg
+        if: ${{ matrix.os != 'windows-latest' }}
+
+      - name: Mac Install Brew
+        run: brew install automake libtool
```
Comment (GordonSmith) [inline .github/workflows/test-platforms.yml:45]: You might need to include "pkg-config" here

File: .github/workflows/test-platforms.yml [hunk 4]
```diff
@@ -7,21 +7,63 @@ on:
...
-        run: cmake --build build/ --target ALL_BUILD
+
+      - name: Generate Solution Windows
+        run: cmake -B build -A Win32 -DCMAKE_TOOLCHAIN_FILE='${{ github.workspace }}/vcpkg/scripts/buildsystems/vcpkg.cmake'
         if: ${{ matrix.os == 'windows-latest' }}
-      - name: Cmake Build all targets on Unix
-        run: cmake --build build/ --target all
+
+      - name: Generate Solution Linux
+        run: cmake -DCMAKE_BUILD_TYPE=Release -Bbuild -S.
```
Comment (GordonSmith) [inline .github/workflows/test-platforms.yml:60]: Should this be ` cmake -DCMAKE_BUILD_TYPE=Release -B build -S .`?

File: CMakeLists.txt [hunk 5]
```diff
@@ -4,9 +4,12 @@ project(nlp VERSION 0.1.0)
 include(CTest)
 enable_testing()
 
-set(CMAKE_GENERATOR_PLATFORM Win32)
+find_package(ICU COMPONENTS uc i18n io REQUIRED)
```
Comment (GordonSmith) [inline CMakeLists.txt:7]: Try:  
```
find_package(ICU REQUIRED
  COMPONENTS uc i18n io
)
```

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: .github/workflows/test-platforms.yml (hunk 4 above)
Comment (dehilsterlexis) [reply .github/workflows/test-platforms.yml:60]: It worked for Linux. It is a bit strange but some coders like "compact". I'm not partial - either way.

File: CMakeLists.txt (hunk 5 above)
Comment (dehilsterlexis) [reply CMakeLists.txt:7]: Yea, I have seen both. I will give it a try.

File: .github/workflows/test-platforms.yml (hunk 3 above)
Comment (dehilsterlexis) [reply .github/workflows/test-platforms.yml:45]: I'm not familiar with that. I will check it out.

File: .github/workflows/test-platforms.yml (hunk 1 above)
Comment (dehilsterlexis) [reply .github/workflows/test-platforms.yml:12]: Tried it. It failed.

**Pull request merged**